File: task-tracker/src/main/java/ru/powergleb/task/tracker/frontend/emailgreeting/services/EmailCreatorService.java
package ru.powergleb.task.tracker.frontend.emailgreeting.services;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import ru.powergleb.task.tracker.frontend.emailgreeting.interfaces.EmailCreator;
import ru.powergleb.task.tracker.frontend.emailgreeting.model.dto.EmailDTO;
import ru.powergleb.task.tracker.frontend.security.model.User;

import java.util.ArrayList;
import java.util.List;

@Service
@RequiredArgsConstructor
public class EmailCreatorService {

    private final EmailCreator emailCreator;

    public List<EmailDTO> getEmailMessages(List<User> users) {
        List<EmailDTO> emailMessages = new ArrayList<>();

        for (User user : users) {
            EmailDTO emailDTO = new EmailDTO();
            emailDTO.setRecipientAddress(emailCreator.createEmailAddress(user));
            emailDTO.setTitle(emailCreator.createEmailTitle(user));
            emailDTO.setText(emailCreator.createEmailText(user));
            emailMessages.add(emailDTO);
        }
        return emailMessages;
    }
}

File: task-tracker-scheduler/src/main/kotlin/ru/powergleb/task/tracker/scheduler/service/EmailCreatorService.kt
package ru.powergleb.task.tracker.scheduler.service

import ru.powergleb.task.tracker.scheduler.interfaces.EmailCreator
import ru.powergleb.task.tracker.scheduler.model.User
import ru.powergleb.task.tracker.scheduler.model.dto.EmailDTO
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class EmailCreatorService(@Autowired private val emailCreator: EmailCreator) {
    fun getEmailMessages(users: MutableList<User>): MutableList<EmailDTO> {
        val emailMessages: MutableList<EmailDTO> = mutableListOf()
        for (user in users) {
            if (user.tasks.size > 0) {
                val emailDTO = EmailDTO(
                    emailCreator.createEmailAddress(user),
                    emailCreator.createEmailTitle(user),
                    emailCreator.createEmailText(user)
                )
                emailMessages.add(emailDTO)
            }
        }
        return emailMessages
    }
}